tests/v2/validation/provisioning/providers.go

```diff
@@ -0,0 +1,59 @@
+package provisioning
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials/aws"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials/digitalocean"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials/harvester"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type CloudCredFunc func(rancherClient *rancher.Client) (*cloudcredentials.CloudCredential, error)
+type MachinePoolFunc func(generatedPoolName, namespace string) *unstructured.Unstructured
+
+type Provider struct {
+	Name            string
+	MachineConfig   string
+	MachinePoolFunc MachinePoolFunc
+	CloudCredFunc   CloudCredFunc
+}
+
+func CreateProvider(name string) Provider {
+	switch {
+	case name == "harvester":
+		provider := Provider{
+			Name:            name,
+			MachineConfig:   machinepools.HarvesterResourceConfig,
+			MachinePoolFunc: machinepools.NewHarvesterMachineConfig,
+			CloudCredFunc:   harvester.CreateHarvesterCloudCredentials,
+		}
+		return provider
+	case name == "aws":
+		provider := Provider{
+			Name:            name,
+			MachineConfig:   machinepools.AWSResourceConfig,
+			MachinePoolFunc: machinepools.NewAWSMachineConfig,
+			CloudCredFunc:   aws.CreateAWSCloudCredentials,
+		}
+		return provider
+	case name == "do":
+		provider := Provider{
+			Name:            name,
+			MachineConfig:   machinepools.DOResourceConfig,
+			MachinePoolFunc: machinepools.NewDigitalOceanMachineConfig,
+			CloudCredFunc:   digitalocean.CreateDigitalOceanCloudCredentials,
+		}
+		return provider
+	//TODO: if no other node driver is found, we should default to custom cluster
+	default:
+		provider := Provider{
```

Comment:
I disagree, I think the default should be a `panic()` along the lines of a type not existing. 
Something like this maybe:
```
default:
            fmt.Printf("Provider:%s not found", name)
```